gdal/swig/python/osgeo/utils/gdal_calc.py

```diff
@@ -117,7 +162,37 @@ def GetOutputDriverFor(filename):
         print("Several drivers matching %s extension. Using %s" % (ext if ext else '', drv_list[0]))
     return drv_list[0]
 
-################################################################
+
+GT_SAME = -1
+GT_ALMOST_SAME = -2
+GT_COMPATIBLE_DIFF = -3
+GT_INCOMPATIBLE_OFFSET = 0
+GT_INCOMPATIBLE_PIXEL_SIZE = 1
+GT_INCOMPATIBLE_ROTATION = 2
+GT_NON_ZERO_ROTATION = 3
+
+
+def gt_diff(gt0, gt1, diff_support=(False, False, False, False), eps=0.0):
+    if isinstance(eps, numbers.Number):
+        eps = (eps, eps, eps)
+    if gt0 == gt1:
+        return -1  # same gt
+    same = (
+        eps[0] <= (abs(gt0[0] - gt1[0]) + abs(gt0[3] - gt1[3])),  # same offset
+        eps[1] <= (abs(gt0[1] - gt1[1]) + abs(gt0[5] - gt1[5])),  # same pixel size
+        eps[2] <= (abs(gt0[2] - gt1[2]) + abs(gt0[4] - gt1[4])),  # same rotation
+        gt0[2] == gt0[4] == gt1[2] == gt1[4] == 0)  # rotation zero
+    if same[0] and same[1] and same[2]:
+        return -2  # almost same gt
+    same = (
+        same[0] or diff_support[0],
```

Comment:
```suggestion
        (same[0] or diff_support[0], GT_INCOMPATIBLE_OFFSET),
```
and similarly for 3 following lines
And then in the for loop you can return the error code from the above tuples instead of relying on the values of the error code being 0,1,2,3

Reply:
OK, I've taken now a similar approach.